ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/KtlintSuppressionRuleTest.kt

```diff
@@ -0,0 +1,798 @@
+                    /* ktlint-disable some-rule-set:some-rule-id */
+                    doSomething()
+                }
+                """.trimIndent()
+            val formattedCode =
+                """
+                @Suppress("aaa", "ktlint:some-rule-set:some-rule-id", "zzz")
+                fun foo() {
+                    doSomething()
+                }
+                """.trimIndent()
+            ktlintSuppressionRuleAssertThat(code)
+                .hasLintViolation(3, 8, "Directive 'ktlint-disable' is deprecated. Replace with @Suppress annotation")
+                .isFormattedAs(formattedCode)
+        }
+
+        @Test
+        fun `Given a ktlint-disable directive for which the target element is already annotated with @SuppressWarnings then add the ktlint suppression and sort all suppressions alphabetically`() {
+            val code =
+                """
+                @SuppressWarnings("aaa", "zzz")
+                fun foo() {
+                    /* ktlint-disable some-rule-set:some-rule-id */
+                    doSomething()
+                }
+                """.trimIndent()
+            val formattedCode =
+                """
+                @SuppressWarnings("aaa", "ktlint:some-rule-set:some-rule-id", "zzz")
+                fun foo() {
+                    doSomething()
+                }
+                """.trimIndent()
+            ktlintSuppressionRuleAssertThat(code)
+                .hasLintViolation(3, 8, "Directive 'ktlint-disable' is deprecated. Replace with @Suppress annotation")
+                .isFormattedAs(formattedCode)
+        }
+
+        @Test
+        fun `Given a ktlint-disable directive for which the target element is already annotated with both @Suppress and @SuppressWarnings then add the ktlint suppression to the @Suppress`() {
+            val code =
+                """
+                @Suppress("aaa", "zzz")
+                @SuppressWarnings("bbb", "yyy")
+                fun foo() {
+                    /* ktlint-disable some-rule-set:some-rule-id */
+                    doSomething()
+                }
+                """.trimIndent()
+            val formattedCode =
+                """
+                @Suppress("aaa", "ktlint:some-rule-set:some-rule-id", "zzz")
+                @SuppressWarnings("bbb", "yyy")
+                fun foo() {
+                    doSomething()
+                }
+                """.trimIndent()
+            ktlintSuppressionRuleAssertThat(code)
+                .hasLintViolation(4, 8, "Directive 'ktlint-disable' is deprecated. Replace with @Suppress annotation")
+                .isFormattedAs(formattedCode)
+        }
+    }
+
+    @Nested
+    inner class `Given a ktlint-enable directive` {
+        @Test
+        fun `Given a ktlint-enable directive matching with a ktlint-disable directive`() {
+            val code =
+                """
+                fun foo() {
+                    /* ktlint-disable some-rule-set:some-rule-id */
```

Comment:
multiple enable disable directives either like below
```
/* ktlint-disable some-rule-set:some-rule-id-1 */
/* ktlint-disable some-rule-set:some-rule-id-2 */
// code
/* ktlint-enable some-rule-set:some-rule-id-1 */
/* ktlint-enable some-rule-set:some-rule-id-2 */
```
or
```
/* ktlint-disable some-rule-set:some-rule-id-1 */
// code 1
/* ktlint-enable some-rule-set:some-rule-id-1 */

/* ktlint-disable some-rule-set:some-rule-id-2 */
// code 2
/* ktlint-enable some-rule-set:some-rule-id-2 */
```

Reply:
Hi, @paul-dingemans here expected output would be `@Suppress("ktlint:some-rule-set:some-rule-id-1", "ktlint-disable some-rule-set:some-rule-id-2")` so basically I thing we have to collect all the suppression in the scope and add those in the single `@Suppress`

Reply:
Yes indeed they should be collected in a single @Suppress. Will have to fix this.